src/core/thread/discover_scanner.cpp

```diff
@@ -113,6 +116,16 @@ otError DiscoverScanner::Discover(const Mac::ChannelMask &aScanChannels,
 
     SuccessOrExit(error = Tlv::AppendTlv(*message, Tlv::kDiscovery, &discoveryRequest, sizeof(discoveryRequest)));
 
+    if (mHasJoinerAdvertisement)
+    {
+        // Append MLE Discovery TLV with sub-TLV (MeshCoP Joiner Advertisement).
```

Comment:
This comment does not seem to be valid here. In the code below it, we are not appending Discovery TLV with a sub-TLV as its value and we are just  appending a `JoinerAdv` TLV to the message.

@lanyuwen did you intend for the `JoinerAdv` TLV to be added as one of the sub-TLVs of the `DiscoveryRequestTlv` or as its own TLV? If the former, then the code need to update the length of enclosing `DiscoveryTlv` (as it now has two sub-tlvs as its value).

Reply:
The intention is to append the TLV as a sub-TLV. I'll get it fixed later, and thanks for pointing this out!

Reply:
fixed